src/pyOpenMS/setup.py

```diff
@@ -236,7 +236,7 @@
         extra_compile_args=extra_compile_args,
         extra_objects=objects,
         extra_link_args=extra_link_args,
-		define_macros=[('BOOST_ALL_NO_LIB', None)] ## Deactivates boost autolink (esp. on win).
+		define_macros=[('BOOST_ALL_NO_LIB', None), ("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")] ## Deactivates boost autolink (esp. on win). Shuts up the damn "deprecated NumPy API" warning spam (https://docs.cython.org/en/latest/src/userguide/numpy_tutorial.html#numpy-compilation)
```

Comment:
setup.py is not used for building the shared libraries anymore. Only to package the wheel.
Building is done with CMake.

Add it to the target compile definitions or whatever suits you here:
https://github.com/OpenMS/OpenMS/blob/c601b60c717da64ed77bae3460b131e1d4572d26/src/pyOpenMS/CMakeLists.txt#L621

Reply:
Like so?:
      target_compile_definitions(pyopenms_${nr} PRIVATE NPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION)

Reply:
Sounds about right. In both if-cases.